### PR TITLE
feat: Use pixel content values from resize observer

### PR DESCRIPTION
**Pull request by dvmoritzschoefl**
Newer versions of chrome can measure the actual pixel values a canvas occupies. This is very handy since those are integer values, and not fractional CSS values.

### CODE REVIEW

**Review by thinkh — APPROVED** [review]
@dvmoritzschoefl Thanks for the refactoring / feature. I applied some additional refactorings. Please take another look. Then we can merge this PR.

**Comment by dvmoritzschoefl** [comment]
> @dvmoritzschoefl Thanks for the refactoring / feature. I applied some additional refactorings. Please take another look. Then we can merge this PR.

yes thats fine you can merge it

File: src/vis/vishooks/hooks/useCanvas.ts [hunk 1]
```diff
@@ -28,13 +28,16 @@ export function useCanvas<ContextId extends '2d' | 'webgl' | 'bitmaprenderer' |
       const observer = new ResizeObserver((entries) => {
         if (entries[0]) {
           const newDimensions = entries[0].contentRect;
+          const { inlineSize, blockSize } = entries[0].devicePixelContentBoxSize[0]!;
```
Comment (puehringer) [inline src/vis/vishooks/hooks/useCanvas.ts:31]: @dvmoritzschoefl What will happen on Safari here? It is not supported and probably null?

https://caniuse.com/?search=devicePixelContentBoxSize

image

Comment (dvmoritzschoefl) [reply src/vis/vishooks/hooks/useCanvas.ts:31]: very likely yes, I will add feature detection and use the old way in this case

Comment (dvmoritzschoefl) [reply src/vis/vishooks/hooks/useCanvas.ts]: @puehringer updated the PR. added a feature check and rounded the value in case its not present

**Pull request merged**